pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -137,6 +150,14 @@ func (n *NodeLabeller) run() error {
 		n.logger.Infof("node-labeller cannot get new labels %s\n", err.Error())
 		return err
 	}
+	n.kvmInfoLabels.Lock()
+	defer n.kvmInfoLabels.Unlock()
```

Comment:
it appears you're only using this value in the prepareLabels() function. Can all of this logic be moved there so we reduce the locked area to the exact functionality that needs the lock.